python/tank/util/user_settings.py

```diff
@@ -43,7 +44,17 @@ def _init_singleton(self):
         # Log the default settings
         logger.debug("Default site: %s" % (self.default_site or "<missing>",))
         logger.debug("Default login: %s" % (self.default_login or "<missing>",))
-        logger.debug("Shotgun proxy: %s" % (self._get_filtered_proxy(self.shotgun_proxy or "<missing>"),))
+
+        self._settings_proxy = self._get_settings_proxy()
+        if self._settings_proxy:
+            logger.debug("Shotgun proxy (from settings): %s" % self._get_filtered_proxy(self._settings_proxy))
+        else:
+            self._system_proxy = self._get_system_proxy()
```

Comment:
i feel we should initialize this variable to None further up to make sure it always exists